File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/kms/src/main/java/com/example/kms/CreateGrant.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
package com.example.kms;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.kms.KmsClient;
import software.amazon.awssdk.services.kms.model.CreateGrantRequest;
import software.amazon.awssdk.services.kms.model.CreateGrantResponse;
import software.amazon.awssdk.services.kms.model.KmsException;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class CreateGrant {

    public static void main(String[] args) {

        final String USAGE = "\n" +
                "Usage:\n" +
                "    <keyId> <granteePrincipal> <operation> \n\n" +
                "Where:\n" +
                "    keyId - the unique identifier for the customer master key (CMK) that the grant applies to. \n\n" +
                "    granteePrincipal - the principal that is given permission to perform the operations that the grant permits. \n\n" +
                "    operation - an operation (for example, Encrypt). \n\n" ;

        if (args.length != 3) {
               System.out.println(USAGE);
               System.exit(1);
         }

        String keyId = args[0];
        String granteePrincipal = args[1];
        String operation = args[2];

        Region region = Region.US_WEST_2;
        KmsClient kmsClient = KmsClient.builder()
                .region(region)
                .build();

        String grantId = createGrant(kmsClient, keyId, granteePrincipal, operation);
        System.out.printf("Successfully created a grant with ID %s%n", grantId);
        kmsClient.close();
    }

    public static String createGrant(KmsClient kmsClient, String keyId, String granteePrincipal, String operation) {

    try {
        CreateGrantRequest grantRequest = CreateGrantRequest.builder()
                .keyId(keyId)
                .granteePrincipal(granteePrincipal)
                .operationsWithStrings(operation)
                .build();

        CreateGrantResponse response = kmsClient.createGrant(grantRequest);
        return response.grantId();

     } catch (KmsException e) {
            System.err.println(e.getMessage());
            System.exit(1);
     }
        return "";
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/kms/src/main/kotlin/com/kotlin/kms/CreateGrant.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.kms

import aws.sdk.kotlin.services.kms.KmsClient
import aws.sdk.kotlin.services.kms.model.CreateGrantRequest
import aws.sdk.kotlin.services.kms.model.GrantOperation
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args: Array<String>) {

    val usage = """
        Usage:
            <keyId> <granteePrincipal> <operation>
        Where:
            keyId - the unique identifier for the AWS KMS key that the grant applies to (for example, xxxxxbcd-12ab-34cd-56ef-1234567890ab).
            granteePrincipal - the principal that is given permission to perform the operations that the grant permits. 
            operation - an operation (for example, Encrypt). 
    """

    if (args.size != 3) {
        println(usage)
        exitProcess(0)
    }

    val keyId = args[0]
    val granteePrincipal = args[1]
    val operation = args[2]
    val grantId = createNewGrant(keyId,granteePrincipal,operation )
    println("Successfully created a grant with ID $grantId")
    }

suspend fun createNewGrant(keyIdVal: String?, granteePrincipalVal: String?, operation: String): String? {

            val operationOb = GrantOperation.fromValue(operation)
            val grantOperationList = ArrayList<GrantOperation>()
            grantOperationList.add(operationOb)

            val request = CreateGrantRequest {
                keyId = keyIdVal
                granteePrincipal = granteePrincipalVal
                operations = grantOperationList
            }

            KmsClient { region = "us-west-2" }.use { kmsClient ->
              val response = kmsClient.createGrant(request)
            return response.grantId
        }
  }
